UniWatch/Controllers/LectureController.cs

```diff
@@ -0,0 +1,297 @@
+        {
+            if(lecture.Class == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, "No class id");
+            }
+
+            if(Request.Files.Count == 0)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, "No files selected");
+            }
+
+            var fileStreams = new List<Stream>(Request.Files.Count);
+
+            foreach(HttpPostedFileBase file in Request.Files)
+            {
+                if(file.ContentLength == 0 /*&& file.ContentLength >= MAX_SIZE*/)
+                {
+                    return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"Selected file {file.FileName} is out of bounds");
+                }
+
+                fileStreams.Add(file.InputStream);
+            }
+
+            _manager.LectureManager.RecordLecture(lecture.Class.Id, fileStreams);
+            // TODO: while the system is creating the lecture and taking attendance, display a load screen
+
+            return RedirectToAction("Index", "Lecture", new { classId = lecture.Class.Id });
+        }
+
+        #endregion
+
+        #region Delete
+
+        [HttpGet]
+        public ActionResult Delete(int lectureId)
+        {
+            var lvm = GetLectureViewModel(lectureId);
+
+            if (lvm == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"No lecture with id {lectureId}");
+            }
+
+            return View(lvm);
+        }
+
+        [HttpPost]
+        [ActionName("Delete")]
+        public ActionResult DeleteConfirm(int lectureId, int classId)
+        {
+            var lecture = _manager.LectureManager.Delete(lectureId);
+
+            if (lecture == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest, $"No lecture given");
+            }
+
+            return RedirectToAction("Index", new { classId = classId });
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        /// <summary>
+        /// Get a report for the class ID
+        /// </summary>
+        /// <param name="classId">The ID of the class</param>
+        /// <returns>All the lectures, students, and attendance for a class</returns>
+        private ReportViewModel GetReportViewModel(int classId)
+        {
+            ReportViewModel report = null;
+            var @class = _manager.ClassManager.GetById(classId);
+
+            // If a class was found,
+            // then add the other required information
+            if(@class != null)
+            {
+                report = new ReportViewModel()
+                {
+                    ClassId = @class.Id,
+                    ClassName = @class.Name,
+                    Lectures = new List<Lecture>(@class.Lectures),
+                    Statuses = new List<AttendanceStatus>(@class.Enrollment.Count)
+                };
+
+                foreach (var lecture in @class.Lectures)
```

Comment:
This loop is potentially O(n^3). It can be simplified. The result is also specific to teachers and should be renamed.